python/grass/pygrass/gis/__init__.py

```diff
@@ -455,9 +454,9 @@ def _write(self, mapsets):
         :param mapsets: a list of mapset's names
         :type mapsets: list
         """
-        with open(self.spath, "wb+") as f:
-            ms = [decode(m) for m in self.location.mapsets()]
-            f.write(b"\n".join([encode(m) for m in mapsets if m in ms]))
+        with open(self.spath, "w") as f:
+            ms = self.location.mapsets()
+            f.write("\n".join([m for m in mapsets if m in ms]))
```

Comment:
Small note (**it is not bug**):

`g.mapsets` module write `\n` on the EOF for every line in the SEARCH_PATH file:

```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets operation=set mapset=landsat

GRASS nc_spm_08_grass7/landsat:~ > g.mapsets operation=add mapset=PERMANENT

GRASS nc_spm_08_grass7/landsat:~ > cat -e grassdata/nc_spm_08_grass7/landsat/SEARCH_PATH 
landsat$
PERMANENT$
```

`VisibleMapset` class instance `_write` method do not write `\n` on the EOF on the last line in the SEARCH_PATH file:

```
GRASS nc_spm_08_grass7/landsat:~ > g.mapsets -l
Available mapsets:
PERMANENT climate_2000_2012 landsat tomas

GRASS nc_spm_08_grass7/landsat:~ > cat /tmp/test.py
#!/usr/bin/env python3

import sys

from grass.pygrass.gis import VisibleMapset

def main():
    vm = VisibleMapset(mapset=sys.argv[1])
    vm.extend(["climate_2000_2012", "tomas"])
    print(vm.read())

if __name__ == '__main__':
    main()

GRASS nc_spm_08_grass7/landsat:~ > python /tmp/test.py $(g.gisenv get="MAPSET")
['landsat', 'PERMANENT', 'climate_2000_2012', 'tomas']

GRASS nc_spm_08_grass7/landsat:~ > cat -e grassdata/nc_spm_08_grass7/landsat/SEARCH_PATH 
landsat$
PERMANENT$
climate_2000_2012$
tomasGRASS nc_spm_08_grass7/landsat:~ >
```
`tomasGRASS nc_spm_08_grass7/landsat:~ >` this line should be (missing `$` which is `\n`)

```
tomas$
GRASS nc_spm_08_grass7/landsat:~ >
```


Fixed with (line 459):

```python
eof = "\n"
f.write(eof.join([m for m in mapsets if m in ms]) + eof)
```

and (line 469):

```python
f.write("%s\n" % mapset)
```